### ISSUE TITLE
有计划支持 .NET Core 吗？

### ISSUE THREAD

**Issue by RockNHawk** [tonyqus/toxy#18]
最近业余时间把一个庞大的应用框架迁移到 .NET Core 2.1 上了，改动比想象中小很多，除了 Mvc 部分微软重写了该起来比较痛苦以外，其它倒还挺顺利的，启动速度缩减到了原来的一半。

Toxy 有计划支持 .NET Core 吗 ？

**Comment by RockNHawk** [comment]
自己下了源码正在迁移中，好了会 pull 上来

**Comment by MovGP0** [comment]
你介意用英文写作吗？

**Comment by tonyqus** [comment]
@MovGP0 Thank you for mentioning this. I agree with you that comments on Github should be written in English. 
The problem is that most Chinese developer doesn't have a good command of English. It's a bit shame. But that's the fact.

**Comment by tonyqus** [comment]
@RockNHawk I'm not sure how would you like to migrate Toxy to .NET core because it depends on a lot of 3rd party library. If these library doesn't support .NET core, you will be blocked.

**Comment by MovGP0** [comment]
Porting to .NET Standard would have many benefits. Mostly that .NET Core applications can use it, ie. in Linux or Docker environments (like I have). 

I agree that the dependencies must be ported first. Some of them are already. The rest should compile on .NET Standard without changes. (ie. [RtfDomParser](https://github.com/SourceCodeBackup/RtfDomParser) only needs a new .csproj file). So I guess it should't be too complicated to fix them and make a pull request.

**Comment by tonyqus** [comment]
@MovGP0 .NET is more popular in Australia than in China. I'm now thinking more like a business owner instead of just developer. I agree that technically it's not so complicated . But a lot of 'free' efforts are still needed. 

My major concern here for porting to .NET core is that there is not much requirement from commercial companies. .NET core for them is still a baby. Most of the companies are still using .Net framework since the existing projects are created long time ago. There is no reason for them to migrate to .NET core just because the technology is ready to use.

**Comment by RockNHawk** [comment]
I already have ported Toxy with most features to .NET Core, Some project does't have .NET Core verison, I migration it or replaced It, I will pull the source code a few day.
@tonyqus @MovGP0

**Comment by RockNHawk** [comment]
I have pushed, https://github.com/RockNHawk/toxy.netcore

**Comment by tonyqus** [comment]
@RockNHawk  Very nice. Thank you for your efforts. I'll review it this week.

**Comment by sgf** [comment]
@RockNHawk
nice work! The  .net community lacks such an atmosphere.

**Comment by RockNHawk** [comment]
@tonyqus Yeah, the library I migrated Only NPOI has a open Git Repository, but I just migrated have not tested, and the NPOI seems have compatibility issue , I needs to fix it at first ?

**Comment by tonyqus** [comment]
I've started to migrate NPOI to .NET core. It will be ready soon
